### PR TITLE
Block casting to an interface marked ComInterfaceType.InterfaceIsIInspectable

### LINKED ISSUE #41621: .NET 5 removal of WinRT support does not correctly reject ComInterfaceType.InterfaceIsIInspectable
Found when adding ComImport-based unit tests to C#/WinRT.  An interface defined with ComInterfaceType.InterfaceIsIInspectable will cast without error, but subsequent interface calls will AV.  According to @jkoritzinsky, the correct behavior is to throw an exception on the cast.

Repro:
[unit test code](https://github.com/microsoft/CsWinRT/blob/ac42ed401389e8f9b3e23d7659037ccabc603395/UnitTest/TestComponentCSharp_Tests.cs#L543)
[cast code, Marshal.GetObjectForIUnknown](https://github.com/microsoft/CsWinRT/blob/efe1842a41df268093b456f2080230a5204b6f19/WinRT.Runtime/ObjectReference.cs#L67)

### PR DESCRIPTION
**Pull request by jkoritzinsky**
During the cast path, nothing checks if an interface is an IInspectable interface. This PR adds that check to ensure that we block usage of ComInterfaceType.InterfaceIsIInspectable now that WinRT support is removed.

Fixes #41621

### CODE REVIEW

File: src/tests/Interop/COM/NETClients/IInspectable/App.manifest [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<assembly manifestVersion="1.0" xmlns="urn:schemas-microsoft-com:asm.v1">
+  <assemblyIdentity
+    type="win32" 
+    name="NetPrimitivesIDispatch"
```
Comment (AaronRobinsonMSFT) [inline src/tests/Interop/COM/NETClients/IInspectable/App.manifest:5]: Update.

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
@AaronRobinsonMSFT I'm assuming we want to backport this to 5.0 once this merges into master?

**Comment by AaronRobinsonMSFT** [comment]
@jkoritzinsky Bah. Boo... I forgot the main branch is now 6. It would be nice if you could try and take it to ship room. Just the product change nothing else. I really consider this a missed scenario for our [WinRT in .NET 5.0](https://github.com/dotnet/runtime/issues/35318) plans. It isn't blocking anyone, but shutting down all those entry points would be ideal.

/cc @jeffschwMSFT @scottj1s

File: src/tests/Interop/COM/ServerContracts/Server.Contracts.cs [hunk 2]
```diff
@@ -324,6 +324,22 @@ public interface IConsumeNETServer
         bool EqualByCCW(object obj);
         bool NotEqualByRCW(object obj);
     }
+
+    [ComVisible(true)]
+    [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    [Guid("3021236a-2a9e-4a29-bf14-533842c55262")]
+    internal interface IInspectableTesting
+    {
+        int Add(int i, int j);
```
Comment (elinor-fung) [inline src/tests/Interop/COM/ServerContracts/Server.Contracts.cs:333]: `IInspectableTesting` doesn't have `Add` on the native side?

**Review by elinor-fung — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
The only failure is on Mono WASM and this code only runs on Windows on CoreCLR. I'm going to merge this in.

**Pull request merged**